mountd/mountd.c

```diff
@@ -189,17 +190,18 @@ static int write_file_safe(char *path, char *buff, size_t length, int mode)
 	ret = 0;
 
 err_out:
+	g_free(buff);
 	g_free(path_tmp);
 	return ret;
 }
 
 static int create_subauth_file(char *path_subauth)
 {
-	char subauth_buf[35];
+	char *subauth_buf;
 	GRand *rnd;
 
 	rnd = g_rand_new();
-	sprintf(subauth_buf, "%d:%d:%d\n", g_rand_int_range(rnd, 0, INT_MAX),
+	subauth_buf = g_strdup_printf("%d:%d:%d\n", g_rand_int_range(rnd, 0, INT_MAX),
```

Comment:
Where is freeing subauth_buf ?

Reply:
write_file_safe does. I mentioned it earlier in this thread.

Reply:
Okay. one question, In general, is it a good way to free a buffer that allocated from caller in callee? otherwise, looks good.

Reply:
Probably not. Although the alternative as I mentioned is:

```
int i = x;
g_free(y);
return i;
```

The other possible alternative that I've never gotten to work is using glib's support for the __cleanup__ attribute. Then it could be kept as a single return statement.

Reply:
@neheb Ah.. I didn't fully understand. You probably update this patch soon ? no issue to apply current patch ?

Reply:
Rebased. Should be ready to go now